pennylane/transforms/optimization/optimization_utils.py

```diff
@@ -60,17 +72,47 @@ def _quaternion_product(q1, q2):
     return stack([qw, qx, qy, qz])
 
 
-def _fuse(angles_1, angles_2):
-    """Perform fusion of two angle sets. Separated out so we can do JIT with conditionals."""
-    # Compute the product of the quaternions
-    qw, qx, qy, qz = _quaternion_product(_zyz_to_quat(angles_1), _zyz_to_quat(angles_2))
+def _singular_quat_to_zyz(q, y_arg):
+    """Compute the ZYZ angles for the singular case of qx = qy = 0"""
+    qw, qx, qy, qz = q
+    z1_arg1 = 2 * (qx * qy + qz * qw)
+    z1_arg2 = 1 - 2 * (qx**2 + qz**2)
+    if y_arg > 0:
+        z1 = arctan2(z1_arg1, z1_arg2)
+        y = z2 = 0.0
+    else:
+        z1 = -arctan2(z1_arg1, z1_arg2)
+        y = np.pi
+        z2 = 0.0
+    return stack([z1, y, z2])
 
-    # Convert the product back into the angles fed to Rot
+
+def _singular_quat_to_zyz_jax(q, y_arg):
+    """Compute the ZYZ angles for the singular case of qx = qy = 0 in a jax
+    JIT compatible manner."""
+    from jax.lax import cond
+
+    qw, qx, qy, qz = q
+    z1_arg1 = 2 * (qx * qy + qz * qw)
+    z1_arg2 = 1 - 2 * (qx**2 + qz**2)
+    return cond(
+        y_arg > 0,
+        lambda z1_arg1, z1_arg2: stack([arctan2(z1_arg1, z1_arg2), 0.0, 0.0]),
+        lambda z1_arg1, z1_arg2: stack([-arctan2(z1_arg1, z1_arg2), np.pi, 0.0]),
+        z1_arg1,
+        z1_arg2,
+    )
+
+
+def _regular_quat_to_zyz(q, y_arg):
+    """Compute the ZYZ angles for the regular case (qx != 0 or qy != 0)"""
+    qw, qx, qy, qz = q
+    z1_arg1 = 2 * (qx * qy + qz * qw)
     z1_arg1 = 2 * (qy * qz - qw * qx)
```

Comment:
should the first `z1_arg1` not be in the regular one here?

Reply:
Oh gosh, yes, thanks! :)